### PR TITLE
use different env var for node port range

### PR DESCRIPTION
**Pull request by pngwn**
On spaces we set `GRADIO_NUM_PORTS` to `1` and a specific port. This make sense because we need to run gradio on that port for it to be accessible.

However, the node server is reusing this variable which can be problematic if multiple gradio applications are being spun up in the same space. Every app needs it own node server, but only ever 1 publicly accessible URL. 

The current demos 'work' right now but you'll actually notice that only the first one is being SSRed, the others are falling back to the SPA because it could not find a port (because the limit of 1 had been exhausted).

This PR introduces a separate env variable to control how many node ports we should limit ourselves to, defaulting to 100, like before. 

This should address this issue and out demo spaces should all correctly SSR.

### CODE REVIEW

File: guides/04_additional-features/09_environment-variables.md [hunk 1]
```diff
@@ -22,7 +22,16 @@ Environment variables in Gradio provide a way to customize your applications and
   export GRADIO_SERVER_NAME="0.0.0.0"
   ```
 
-### 3. `GRADIO_ANALYTICS_ENABLED`
+### 3. `GRADIO_NUM_PORTS`
```
Comment (abidlabs) [inline guides/04_additional-features/09_environment-variables.md:25]: Ah thanks!

File: guides/04_additional-features/09_environment-variables.md [hunk 2]
```diff
@@ -22,7 +22,16 @@ Environment variables in Gradio provide a way to customize your applications and
...
+
+- **Description**: Defines the number of ports to try when starting the Gradio server.
+- **Default**: `100`
+- **Example**:
+  ```bash
+  export GRADIO_NUM_PORTS=200
+  ```
+
```
Comment (pngwn) [inline guides/04_additional-features/09_environment-variables.md:33]: This wasn't documented for some reason, added it here.

File: guides/04_additional-features/09_environment-variables.md [hunk 3]
```diff
@@ -136,7 +145,23 @@ Environment variables in Gradio provide a way to customize your applications and
   export GRADIO_EXAMPLES_CACHE="custom_cached_examples/"
   ```
 
+### 16. `GRADIO_NODE_SERVER_NAME`
 
+- **Description**: Defines the host name for the Gradio node server.
```
Comment (abidlabs) [inline guides/04_additional-features/09_environment-variables.md:150]:
```suggestion
- **Description**: Defines the host name for the Gradio node server. (Only applies if `ssr_mode` is set to `True`.)
```

**Comment by abidlabs** [comment]
nice looks good now!

**Comment by pngwn** [comment]
thaaaanks!

**Pull request merged**